inc/taskjoblog.class.php

```diff
@@ -194,17 +185,22 @@ function getState($states_id) {
    static function getStateItemtype($taskjoblogs_id) {
       global $DB;
 
-      $query = "SELECT * FROM glpi_plugin_fusioninventory_taskjobstates
-         LEFT JOIN `glpi_plugin_fusioninventory_taskjoblogs`
-            ON `plugin_fusioninventory_taskjobstates_id`=".
-               "`glpi_plugin_fusioninventory_taskjobstates`.`id`
-         WHERE `glpi_plugin_fusioninventory_taskjoblogs`.`id`='".$taskjoblogs_id."'
-         LIMIT 1";
-      $result=$DB->query($query);
-      while ($data=$DB->fetch_array($result)) {
-         return $data["itemtype"];
+      $params = ['FROM'   => 'glpi_plugin_fusioninventory_taskjobstates',
+                 'LEFT JOIN' => ['glpi_plugin_fusioninventory_taskjoblogs',
+                     ['FKEY' => ['glpi_plugin_fusioninventory_taskjoblogs'   => 'plugin_fusioninventory_taskjobstates_id',
+                                 'glpi_plugin_fusioninventory_taskjobstates' => 'id']]
+                     ],
+                 'FIELDS' => ['itemtype'],
+                 'WHERE'  => ['id' => $taskjoblogs_id],
```

Comment:
This can't works, we don't know of what table the id is searched

Reply:
well it seems that this method is only used in hook.php for search options, but I cannot understand why it's still here...